pkg/authenticator/ecrsecret.go

```diff
@@ -52,23 +51,10 @@ func (s *ecrSecret) AddToConfigMap(ctx context.Context, name string, namespace s
 	if err != nil {
 		return err
 	}
-	var values []string
-	if data, ok := cm.Data[namespace]; ok {
-		values = strings.Split(data, ",")
-	}
-	values = append(values, name)
-	result := make([]string, 0, len(values))
-	check := map[string]bool{}
-	for _, val := range values {
-		_, exists := check[val]
-		if !exists {
-			check[val] = true
-			result = append(result, val)
-		}
-	}
 
-	update := strings.Join(result, ",")
-	cm.Data[namespace] = update
+	css := csset.NewCSSet(cm.Data[namespace])
```

Comment:
Add and delete creates the instances differently 
csset.NewCSSet vs pkg.NewCSSet.

Reply:
Fixed